h/oauth/schemas.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+import colander
+
+from h import i18n
+from h.models.auth_client import GrantType, ResponseType
+from h.schemas.base import CSRFSchema
+
+_ = i18n.TranslationString
+
+
+def enum_schema_node(enum_cls, **kwargs):
+    """
+    Return a `colander.SchemaNode` for a field with a given enum type.
+
+    :param enum_cls: The enum class
+    :type enum_cls: enum.Enum
+    """
+    values = []
+    for name, _ in enum_cls.__members__.items():
+        values.append(name)
+
+    return colander.SchemaNode(colander.String(),
+                               validator=colander.OneOf(values),
```

Comment:
This confused me earlier too. Fields are required unless a default is specified via a `missing` kwarg to the `SchemaNode` constructor.